app/lib/service/security_advisories/backend.dart

```diff
@@ -212,6 +212,28 @@ List<String> sanityCheckOSV(OSV osv) {
     errors.add('Invalid id, id too long (over 255 characters).');
   }
 
+  if (osv.affected == null || osv.affected!.isEmpty) {
+    errors.add('No affected packages for advisory ${osv.id}');
+  } else {
+    bool noAffectedHasVersion = true;
+    osv.affected!
+        .where((a) => a.package.ecosystem.toLowerCase() == 'pub')
+        .forEach((affected) {
+      if (affected.versions == null || affected.versions!.isEmpty) {
+        logger.warning('No versions specified for affected package '
+            '${affected.package.name} in advisory ${osv.id}.');
```

Comment:
Is it wrong for `osv.affected` to contain multiple objects talking about the same package where one has `.versions` and one has `.ranges`?

Keep in mind that it's possible that `affected.package.name` appears multiple times.

Reply:
This should at least not happen for the advisories we import from OSV.dev. And if there is extra information in the one with `ranges` we will take that into consideration.